File: projects/order-creation/core/src/main/java/com/robinfood/core/dtos/transactionrequestdto/CompanyDTO.java
package com.robinfood.core.dtos.transactionrequestdto;

import java.io.Serializable;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CompanyDTO implements Serializable {

    private static final long serialVersionUID = 8925238570926632453L;

    @NotBlank
    private String currency;

    @NotNull
    private Long id;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/CompanyDTO.kt
package com.robinfood.core.dtos.transactionrequest

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class CompanyDTO (

    @NotBlank(message = "Company currency should not be null")
    val currency: String,

    @NotNull(message = "Company id should not be null")
    val id: Long
)
